### DISCUSSION TITLE
Metrics Improvement: Ability for Monitoring and Ability for Exploring Issues and Link to Codebases

### DISCUSSION THREAD

**Discussion by clean99** [autometrics-dev/autometrics-shared#68] [Ideas]
The metrics to codebase link enable users to look at the metrics first, which are more readable. So I would prefer the way to monitor metrics, discover issues on the dashboard as far as possible, and then jump into the codebase.

I personally utilize metrics like this:
1. Observing metrics changing(could be the spike or drop).
2. Local the places that cause the change.
3. Dive deep into the codebase to see what's going on.

I saw this project got useful links from functions to metrics, which are great. But IMHO, the codebase is mixed with lots of implementation details, which are not the NO.1 material to read(although the readability of the codebase is important).

1. Ideally, I can monitor something like the overall metrics, and then I see some functions latency spike, I can: 1. check the function in the codebase. 2. continue exploring the issues on the metrics(eg. click on the point, and I can see the children nodes data, which could help me to find which child caused this issue, recursively). That will be way faster to locate problems for an engineer.

2. Also, I think it will be great to have the ability to compare different functions(or the same function with different versions) performance on the metrics. Now it seems to only have a time-based comparison to the same function.

3. We can display some important information(or at least allow users to customize by themself) on metrics, like the most time-consuming function name under the function that we are checking.

**Comment by IvanMerrill** [comment]
These are all good points, and are generally the way autometrics is being developed. In response to your points specifically:

1. Yes this is how the workflow would happen and viewing the metrics is what you would use prometheus for. There's also the grafana dashboards that can be used to do exactly this already as well. 
2. This is a great point. I suggest this becomes an idea in of itself. 
3. Again, this is what you would use prometheus or grafana for.
